yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,601 @@
+            'ext': 'mp4',
+            'title': 'Mitt Romney\'s Crazy Solution To Climate Change',
+            'thumbnail': 're:https://img.production.rokfin.com/.+',
+            'upload_date': '20211023',
+            'timestamp': 1634998029,
+            'creator': 'Jimmy Dore',
+            'channel': 'Jimmy Dore',
+            'channel_id': 65429,
+            'channel_url': 'https://rokfin.com/TheJimmyDoreShow',
+            'duration': 213.0,
+            'availability': 'public',
+            'is_live': False,
+            'was_live': False,
+            'live_status': 'not_live'
+        }
+    }]
+
+    def _real_extract(self, url_from_user):
+        video_id = self._match_id(url_from_user)
+        downloaded_json = self._download_json(self._META_DATA_BASE_URL + video_id, video_id, note='Downloading video metadata', fatal=False) or {}
+
+        def videoAvailability(y, dic):
+            video_availability = dic['premiumPlan']
+            # 0 - public
+            # 1 - premium only
+
+            if video_availability not in (0, 1):
+                y.report_warning(f'unknown availability code: {video_availability}. The extractor should be fixed')
+                return
+
+            return y._availability(
+                is_private=False,
+                needs_premium=(video_availability == 1),
+                needs_subscription=False,
+                needs_auth=False,
+                is_unlisted=False)
+
+        video_formats_url = url_or_none(traverse_obj(downloaded_json, ('content', 'contentUrl')))
+        availability = try_get(self, lambda x: videoAvailability(x, downloaded_json))
+
+        if video_formats_url:
+            # Before the adoption of M3U, Rokfin stored videos directly in mp4 files:
+            if not(video_formats_url.endswith('.m3u8')):
+                return self.url_result(url=video_formats_url, video_id=video_id, original_url=url_from_user)
+
+            frmts = self._extract_m3u8_formats(m3u8_url=video_formats_url, video_id=video_id, fatal=False)
+            self._sort_formats(frmts)
+        else:
+            frmts = None
+
+        if not frmts:
+            if availability == 'premium_only':
+                self.raise_no_formats(msg='premium content', video_id=video_id, expected=True)
+            elif video_formats_url:
+                self.raise_no_formats(msg='unable to download: missing meta data', video_id=video_id, expected=True)
+            else:
+                # We don't know why there is no (valid) meta data present.
+                self.raise_no_formats(msg='unable to download: don\'t know where to find meta data', video_id=video_id, expected=True)
+
+        content_subdict = lambda key: traverse_obj(downloaded_json, ('content', key))
+        created_by = lambda key: traverse_obj(downloaded_json, ('createdBy', key))
+        channel_name = traverse_obj(downloaded_json, ('createdBy', 'name'))
+        downloaded_json_get = downloaded_json.get
```

Comment:
this may have looked like a smart thing when coding but it's really not.
Makes the maintainability and especially reviewing hard.

There's nothing wrong in passing same key (say `content`) in multiple fields.

Only good reason to extract out a sub-dict is when the key lengths are big and the return line becomes long.
if that's the case do 
`content_subdict = download_json.get('content') or {}`
and so on